src/main/java/com/zendesk/maxwell/MaxwellConfig.java

```diff
@@ -57,31 +66,89 @@ public void setInitialPosition(BinlogPosition position) throws IOException {
 		}
 	}
 
-	public static MaxwellConfig fromPropfile(String filename) throws IOException {
+	private void parseOptions(String [] argv) {
+		OptionParser parser = new OptionParser();
+		parser.accepts( "host" ).withRequiredArg();
+		parser.accepts( "password" ).withRequiredArg();
+		parser.accepts( "user" ).withRequiredArg();
+		parser.accepts( "port" ).withRequiredArg();
+		parser.accepts( "producer" ).withRequiredArg();
+
+		OptionSet options = parser.parse(argv);
+
+		if ( options.has("host"))
+			this.mysqlHost = (String) options.valueOf("host");
+		if ( options.has("password"))
+			this.mysqlPassword = (String) options.valueOf("password");
+		if ( options.has("user"))
+			this.mysqlUser = (String) options.valueOf("user");
+		if ( options.has("port"))
+			this.mysqlPort = Integer.valueOf((String) options.valueOf("port"));
+		if ( options.has("producer"))
+			this.producer = (String) options.valueOf("producer");
+	}
+
+	private void parseFile(String filename) throws IOException {
 		Properties p = new Properties();
-		FileReader reader = new FileReader(new File(filename));
+		File file = new File(filename);
+		if ( !file.exists() )
+			return;
+
+		FileReader reader = new FileReader(file);
 		p.load(reader);
 
-		MaxwellConfig config = new MaxwellConfig();
 
-		config.mysqlHost     = p.getProperty("host", "127.0.0.1");
-		config.mysqlPassword = p.getProperty("password");
-		config.mysqlUser     = p.getProperty("user");
-		config.mysqlPort     = Integer.valueOf(p.getProperty("port", "3306"));
+		this.mysqlHost     = p.getProperty("host", "127.0.0.1");
+		this.mysqlPassword = p.getProperty("password");
+		this.mysqlUser     = p.getProperty("user");
+		this.mysqlPort     = Integer.valueOf(p.getProperty("port", "3306"));
 
-		config.currentPositionFile = p.getProperty("position_file", "maxwell.position");
-		config.kafkaProperties = new Properties();
+		this.currentPositionFile = p.getProperty("position_file");
+		this.kafkaProperties = new Properties();
+		this.producer      = p.getProperty("producer");
+		this.outputFile      = p.getProperty("output_file");
 
 		for ( Enumeration<Object> e = p.keys(); e.hasMoreElements(); ) {
 			String k = (String) e.nextElement();
 			if ( k.startsWith("kafka.")) {
-				config.kafkaProperties.setProperty(k.replace("kafka.", ""), p.getProperty(k));
+				this.kafkaProperties.setProperty(k.replace("kafka.", ""), p.getProperty(k));
 			}
 		}
+
+	}
+
+	public static MaxwellConfig buildConfig(String filename, String [] argv) throws IOException {
+		MaxwellConfig config = new MaxwellConfig();
+
+		config.parseFile(filename);
+		config.parseOptions(argv);
+		config.setDefaults();
```

Comment:
Tough to understand precedence here, you would expect defaults to be at the top.